### PR TITLE
HPCC-12839 Fix crash when row being looked up in a dictionary is aliased

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
(Example sbfe4 in private suite triggered it)

**Comment by ghalliday** [comment]
@jamienoss please review.

File: ecl/hqlcpp/hqlcset.cpp [hunk 1]
```diff
@@ -817,6 +817,16 @@ InlineLinkedDictionaryCursor::InlineLinkedDictionaryCursor(HqlCppTranslator & _t
 {
 }
 
+IHqlExpression * InlineLinkedDictionaryCursor::getFirstSearchValue(IHqlExpression * searchExpr, IHqlExpression * searchRecord)
+{
+    if (searchExpr->getOperator() == no_alias)
+        searchExpr = searchExpr->queryChild(0);
+
+    IHqlExpression * matched = getExtractSelect(searchExpr->queryChild(0), queryFirstField(searchRecord), false);
```
Comment (jamienoss) [inline ecl/hqlcpp/hqlcset.cpp]: Is it within the semantics of this function to hardwire bool okToSkipRow = false, or should it be passed in?

**Comment by jamienoss** [comment]
Regarding example ecl, the jira suggests that this was triggered from an existing example (?). Since it crashed I'm assuming that the regression suit wasn't detecting this or that the specific code was not being executed. Are changes required to existing example and/or new example?

**Comment by jamienoss** [comment]
@ghalliday back to you.

File: ecl/hqlcpp/hqlcset.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hqlcpp/hqlcset.cpp]: Yes it is within the semantics of this function - but in fact there will never be a skip in the transform anyway, so it wouldn't matter either way.

**Comment by ghalliday** [comment]
It was triggered from a new example.  I have a bit more of an idea of how it is being triggered and could possibly create a simple example that can go in the public regression suite.

**Comment by ghalliday** [comment]
@richardkchapman main question for this is which version should it target?  I think it should target 5.0.6, but should we targeted even earlier given the systems they are seeing the problems on.

File: ecl/hqlcpp/hqlcset.cpp [hunk 2]
```diff
@@ -817,6 +817,16 @@ InlineLinkedDictionaryCursor::InlineLinkedDictionaryCursor(HqlCppTranslator & _t
 {
 }
 
+IHqlExpression * InlineLinkedDictionaryCursor::getFirstSearchValue(IHqlExpression * searchExpr, IHqlExpression * searchRecord)
+{
+    if (searchExpr->getOperator() == no_alias)
```
Comment (richardkchapman) [inline ecl/hqlcpp/hqlcset.cpp]: Can you ever have an alias of an alias?

**Comment by ghalliday** [comment]
Retargeted in 4.2.12 https://github.com/hpcc-systems/HPCC-Platform/pull/6881

**Pull request closed without merging**